### PR TITLE
OCPBUGS-7282: Node Exporter ignores network interface under name "cali[a-f0-9]*"

### PR DESCRIPTION
**Pull request by raptorsun**
* [x] I added CHANGELOG entry for this change.
* [ ] No user facing changes, so no entry in CHANGELOG was needed.

This PR resolves https://issues.redhat.com/browse/OCPBUGS-7282 .
The issue is similar to https://issues.redhat.com/browse/OCPBUGS-2729 .

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@
 - [#1888](https://github.com/openshift/cluster-monitoring-operator/pull/1888) Add nodeExporter.collectors.netdev settings.
 - [#1884](https://github.com/openshift/cluster-monitoring-operator/issues/1882) Allow configuring secrets in alertmanager component (UWM)
 - [#1893](https://github.com/openshift/cluster-monitoring-operator/pull/1893) Add nodeExporter.collectors.netclass settings.
+- [#xxxx](https://github.com/openshift/cluster-monitoring-operator/pull/xxxx) Node Exporter ignores network interface under name "cali[a-f0-9]*".
```
Comment (jan--f) [inline CHANGELOG.md:15]:
```suggestion
- [#1905](https://github.com/openshift/cluster-monitoring-operator/pull/1905) Node Exporter ignores network interface under name "cali[a-f0-9]*".
```

**Review by jan--f — CHANGES_REQUESTED** [review]
Only the changelog needs an update. Otherwise :+1:

File: jsonnet/main.jsonnet [hunk 2]
```diff
@@ -190,17 +190,19 @@ local inCluster =
         },
         // NOTE:
         // "ignoredNetworkDevices" sets the 2 arguments "--collector.netclass.ignored-devices" and "--collector.netdev.device-exclude".
-        // 5 kinds of virtual NICs will be ignored:
+        // 6 kinds of virtual NICs will be ignored:
         // 1. veth network interface associated with containers.
         // 2. OVN renames veth.* to <rand-hex>@if<X> where X is /sys/class/net/<if>/ifindex
         // thus [a-z0-9]{15}}
```
Comment (simonpasquier) [inline jsonnet/main.jsonnet:196]:
```suggestion
        // thus [a-z0-9]{15}
```

**Comment by raptorsun** [comment]
Thanks for the review @jan--f @simonpasquier  :) 
I have removed the entry in the change log alongside the correction on the typo since it is tracked in JIRA already.

**Comment by juzhao** [comment]
@raptorsun launched bot cluster with PR
`launch openshift/cluster-monitoring-operator#1905 gcp`
but did not see `collector.netclass.ignored-devices` and `collector.netdev.device-exclude` settings in node-exporter, full info see: http://pastebin.test.redhat.com/1093428

and checked in 4.13.0-0.nightly-2023-02-27-101545, there is `collector.netclass.ignored-devices` setting
see from: http://pastebin.test.redhat.com/1093429

**Comment by raptorsun** [comment]
> 



> @raptorsun launched bot cluster with PR `launch openshift/cluster-monitoring-operator#1905 gcp` no configuration for node-exporter, did not see `collector.netclass.ignored-devices` and `collector.netdev.device-exclude` settings in node-exporter, full info see: http://pastebin.test.redhat.com/1093428
> 
> and checked in 4.13.0-0.nightly-2023-02-27-101545, no configuration for node-exporter, there is `collector.netclass.ignored-devices` setting see from: http://pastebin.test.redhat.com/1093429
> 
> after disabled netdev/netclass on bot cluster, would see `collector.netclass.ignored-devices` and `collector.netdev.device-exclude` settings in node-exporter, since OCPBUGS-7282 is filed on 4.12, disable netdev/netclass is not supported then, I don't think we should disable netdev/netclass
> 
> ```
> $ oc -n openshift-monitoring get ds node-exporter -oyaml | grep collector
>         - --no-collector.wifi
>         - --collector.filesystem.mount-points-exclude=^/(dev|proc|sys|run/k3s/containerd/.+|var/lib/docker/.+|var/lib/kubelet/pods/.+)($|/)
>         - --collector.netclass.ignored-devices=^(veth.*|[a-f0-9]{15}|enP.*|ovn-k8s-mp[0-9]*|br-ex|br-int|br-ext|br[0-9]*|tun[0-9]*|cali[a-f0-9]*)$
>         - --collector.netdev.device-exclude=^(veth.*|[a-f0-9]{15}|enP.*|ovn-k8s-mp[0-9]*|br-ex|br-int|br-ext|br[0-9]*|tun[0-9]*|cali[a-f0-9]*)$
>         - --collector.cpu.info
>         - --collector.textfile.directory=/var/node_exporter/textfile
>         - --no-collector.cpufreq
>         - --no-collector.tcpstat
>         - --no-collector.netdev
>         - --no-collector.netclass
>         - --no-collector.buddyinfo
>         - '[[ ! -d /node_exporter/collectors/init ]] || find /node_exporter/collectors/init
> ```

Thank you for raising the issue.
I created [a ticket for this bug](https://issues.redhat.com/browse/OCPBUGS-8215) and will create a patch soon.

**Comment by raptorsun** [comment]
https://github.com/openshift/cluster-monitoring-operator/pull/1909

Hello @juzhao , the bug fix has been created :) 
It would be great if we can merge it before code freeze.

**Comment by juzhao** [comment]
issue is fixed, details see from bug comment
/label qe-approved

**Comment by simonpasquier** [comment]
/hold

waiting for https://github.com/openshift/cluster-monitoring-operator/pull/1909

**Review by simonpasquier — COMMENTED** [review]
/lgtm
/hold cancel

**Pull request merged**